plugins/inputs/vsphere/README.MD

```diff
@@ -0,0 +1,350 @@
+vsphere_vm_net,esxhostname=DC0_H0,guest=other,host=host.example.com,moid=vm-38,os=Mac,source=DC0_H0_VM1,vcenter=localhost:8989,vmname=DC0_H0_VM1 bytesRx_average=215i,bytesTx_average=275i 1535660339000000000
+vsphere_vm_virtualDisk,esxhostname=DC0_H0,guest=other,host=host.example.com,moid=vm-38,os=Mac,source=DC0_H0_VM1,vcenter=localhost:8989,vmname=DC0_H0_VM1 write_average=252i,read_average=14i 1535660339000000000
+vsphere_vm_cpu,clustername=DC0_C0,esxhostname=DC0_C0_H0,guest=other,host=host.example.com,moid=vm-41,os=Mac,source=DC0_C0_RP0_VM0,vcenter=localhost:8989,vmname=DC0_C0_RP0_VM0 usage_average=8,used_summation=2183i,demand_average=354i,run_summation=3542i,ready_summation=128i 1535660339000000000
+vsphere_vm_net,clustername=DC0_C0,esxhostname=DC0_C0_H0,guest=other,host=host.example.com,moid=vm-41,os=Mac,source=DC0_C0_RP0_VM0,vcenter=localhost:8989,vmname=DC0_C0_RP0_VM0 bytesRx_average=178i,bytesTx_average=200i 1535660339000000000
+vsphere_vm_virtualDisk,clustername=DC0_C0,esxhostname=DC0_C0_H0,guest=other,host=host.example.com,moid=vm-41,os=Mac,source=DC0_C0_RP0_VM0,vcenter=localhost:8989,vmname=DC0_C0_RP0_VM0 write_average=283i,read_average=12i 1535660339000000000
+vsphere_vm_cpu,clustername=DC0_C0,esxhostname=DC0_C0_H0,guest=other,host=host.example.com,moid=vm-44,os=Mac,source=DC0_C0_RP0_VM1,vcenter=localhost:8989,vmname=DC0_C0_RP0_VM1 demand_average=328i,run_summation=3481i,ready_summation=122i,usage_average=7.95,used_summation=2167i 1535660339000000000
+vsphere_vm_net,clustername=DC0_C0,esxhostname=DC0_C0_H0,guest=other,host=host.example.com,moid=vm-44,os=Mac,source=DC0_C0_RP0_VM1,vcenter=localhost:8989,vmname=DC0_C0_RP0_VM1 bytesTx_average=282i,bytesRx_average=196i 1535660339000000000
+vsphere_vm_virtualDisk,clustername=DC0_C0,esxhostname=DC0_C0_H0,guest=other,host=host.example.com,moid=vm-44,os=Mac,source=DC0_C0_RP0_VM1,vcenter=localhost:8989,vmname=DC0_C0_RP0_VM1 write_average=321i,read_average=13i 1535660339000000000
+vsphere_host_disk,esxhostname=DC0_H0,host=host.example.com,moid=host-19,os=Mac,source=DC0_H0,vcenter=localhost:8989 read_average=39i,write_average=2635i 1535660339000000000
+vsphere_host_disk,disk=/var/folders/rf/txwdm4pj409f70wnkdlp7sz80000gq/T/govcsim-DC0-LocalDS_0-367088371@folder-5,esxhostname=DC0_H0,host=host.example.com,moid=host-19,os=Mac,source=DC0_H0,vcenter=localhost:8989 write_average=2635i,read_average=30i 1535660339000000000
+vsphere_host_mem,esxhostname=DC0_H0,host=host.example.com,moid=host-19,os=Mac,source=DC0_H0,vcenter=localhost:8989 usage_average=98.5 1535660339000000000
+vsphere_host_net,esxhostname=DC0_H0,host=host.example.com,moid=host-19,os=Mac,source=DC0_H0,vcenter=localhost:8989 usage_average=1887i,bytesRx_average=662i,bytesTx_average=251i 1535660339000000000
+vsphere_host_net,esxhostname=DC0_H0,host=host.example.com,interface=vmnic0,moid=host-19,os=Mac,source=DC0_H0,vcenter=localhost:8989 usage_average=1481i,bytesTx_average=899i,bytesRx_average=992i 1535660339000000000
```

Comment:
The two lines above, L335:336, is one of them meant to be an aggregate metric of all interfaces on the host?  If so it would be ideal if we could, at least by default, collect only the total usage or the per interface metrics.

Reply:
That's what the [host|vm|cluster|datastore]_instances does. If you set it to false, you'll get the behavior you're asking for. By default, they're set to "true", except for datastores. This reflects some kind of "best practice" of what people usually want to collect.

Reply:
Do you think we should add support for collecting only the instance metrics?  If you collect instance metrics then there is less reason to store the totals since it can be computed at query time.  Perhaps the instance variables could be something like: `vm_metric_mode = "instance|aggregate|all".

One thing that is a moderate issue with metrics tagged like this is selecting only the instance without the totals.  For example say we have these metrics:
```
vsphere_host_net,interface=vmnic0 bytesTx_average=1i
vsphere_host_net,interface=vmnic1 bytesTx_average=3i
vsphere_host_net bytesTx_average=2i
```

To average only the instance versions you need to know the secret method to remove series containing the total aggregation:
```
select mean(bytesTx_average) from vsphere_host_net where interface =~ /./
```

What we usually do is try to rename the field by appending the aggregation type, the new field name ensures the values do not get aggregated together:
```
vsphere_host_net,interface=vmnic0 bytesTx_average=1i
vsphere_host_net,interface=vmnic1 bytesTx_average=3i
vsphere_host_net bytesTx_average_average=2i
```

So that would be very ugly, maybe something like this would be better?:
```
vsphere_host_net,interface=vmnic0 bytesTx_average=1i
vsphere_host_net,interface=vmnic1 bytesTx_average=3i
vsphere_host_net total_bytesTx_average=2i
```

Reply:
I see your point that it's hard to search to things that are missing a certain point tag. The problem is that it's tricky to reliably determine whether a metric in vCenter will ever show up with an instance metric, so you'd have to add the "total" or "average" to anything that's not a metric on an instance. You never know if that same metric is going to show up in the future with an instance. @puckpuck what do you think?

Reply:
E.g. `cpu.blablabla` doesn't have a metric in vSphere version N, so all metrics are reported as just `cpu.blablabla`. In version N+1, the metric is given an instance and all of a sudden `cpu.blablabla` becomes `cpu.blablabla.total`. I really don't like that.

Reply:
Yeah, that would be a downside, you don't want to switch your dashboards because you change this minor setting.  It seems to me that the best solution would be to collect only one type or the other, at least by default.  Don't have to worry about excluding data from the query if you don't have it, and I think having both forms is redundant anyway.

Reply:
I think the best solution is to always emit the instance tag and send it as "all" for the aggregate metrics (i.e. metrics without an instance tag).

Reply:
That would be similar to the way e.g. input.cpu does it.

Reply:
That will work.

Reply:
Ok. Commit coming shortly.